website/docs/basics/blocks-component-in-block.md

```diff
@@ -342,3 +342,303 @@ echo wp_kses_post(
+```js
+import React from 'react';
+import { props } from '@eightshift/frontend-libs/scripts/editor';
+import { HeadingEditor as HeadingEditorComponent } from '../../../components/heading/components/heading-editor';
+import manifest from './../manifest.json';
+
+export const HeadingEditor = ({attributes, setAttributes}) => {
+  const {
+    blockName,
+  } = manifest;
+
+  const {
+    blockClass,
+  } = attributes;
+
+  return (
+    <div className={blockClass}>
+      <HeadingEditorComponent
+        setAttributes={setAttributes}
+        {...props(attributes, blockName, '', true)}
+      />
+    </div>
+  );
+};
+```
+
+In JavaScript, you spread the results of the props helper.
+
+**Block heading.php:**
+```php
+<?php
+
+/**
+ * Template for the Heading Block view.
+ *
+ * @package Redesign
+ */
+
+use Redesign\Blocks\Blocks;
+use RedesignVendor\EightshiftLibs\Helpers\Components;
+
+$manifest = Components::getManifest(__DIR__);
+$blockName = $attributes['blockName'] ?? $manifest['blockName'];
+
+$blockClass =  Components::checkAttr('blockClass', $attributes, $manifest);
+
+?>
+
+<div class="<?php echo esc_attr($blockClass); ?>">
+  <?php
+  echo Components::render( // phpcs:ignore WordPress.Security.EscapeOutput.OutputNotEscaped
+    'heading',
+    Blocks::props($attributes, $blockName, '', true)
+  );
+  ?>
+</div>
+```
+
+In PHP you just provide the results of the props helper. As you can see in the [props helper docs](helpers-javascript#props) you must provide the fourth parameter to be `true` to distinguish is this block or component.
+
+Now let's see how the `Component heading` looks like.
+
+**Component Heading manifest.json:**
+```json
+{
+  "componentName": "heading",
+  "components": {
+    "heading": "typography"
+  },
+}
+```
+
+In `Component heading` we are using the `typography` component but we are not using the default name but rather we are changing the attribute names from `typography` to `heading`.
+
+**Component heading-editor.js:**
+
+*The same is for options or toolbar components!*
+
+```js
+import React from 'react';
+import { props } from '@eightshift/frontend-libs/scripts/editor';
+import { TypographyEditor } from './../../typography/components/typography-editor';
+import manifest from './../manifest.json';
+ 
+export const HeadingEditor = (attributes) => {
+  const {
+    setAttributes,
+    componentName = manifest.componentName,
+    blockClass,
+  } = attributes;
+
+  return (
+    <>
+      <TypographyEditor
+        selectorClass={componentName}
+        blockClass={blockClass}
+        setAttributes={setAttributes}
+        {...props(attributes, 'typography', componentName)}
+      />
+    </>
+  );
+};
+```
+
+The difference here is that you don't need to provide the fourth parameter here because this is a component. In this example, you are swiping attribute names so we must provide the target component name as a `second parameter and the current component name as a `third` parameter.
```

Comment:
```suggestion
The difference here is that you don't need to provide the fourth parameter because this is a component. In this example, you are swapping attribute names so we must provide the target component name as a `second` parameter and the current component name as a `third` parameter.
```